packages/devtools_app/lib/src/memory/flutter/memory_heatmap.dart

```diff
@@ -240,19 +285,50 @@ class FlameChartRenderObject extends RenderBox {
     markNeedsPaint();
   }
 
+  List<String> _exposeMatchNames(String searchName) {
+    return matchNodeNames(_sizes.root.children, searchName);
+  }
+
+  /// Look for the node with a particular name (depth first traversal).
+  List<String> matchNodeNames(Map<String, Node> children, String searchName) {
+    final matches = <String>[];
+    final matchName = searchName.toLowerCase();
+    for (var child in children.entries) {
+      final node = child.value;
+      if (node.children.isEmpty) {
+        final lcNodeName = node.name.toLowerCase();
+        if (!lcNodeName.endsWith('.dart') && lcNodeName.startsWith(matchName)) {
+          matches.add(node.name);
+        }
+      } else {
+        final lcNodeName = node.name.toLowerCase();
+        if (!lcNodeName.endsWith('.dart') && lcNodeName.startsWith(matchName)) {
+          matches.add(node.name);
+        }
```

Comment:
this code is repeated. you could rewrite this as:
```
final node = child.value;
final lcNodeName = node.name.toLowerCase();
if (!lcNodeName.endsWith('.dart') && lcNodeName.startsWith(matchName)) {
  matches.add(node.name);
}
if (node.children.isNotEmpty) {
  // child matches code...
}
```

Reply:
Done.